jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,460 @@
+        self.waveband_selected = None
+
+        self._full_registry_results = None
+        self.resource_selected = None
+        self.resource = SelectPluginComponent(
+            self, items="resource_choices", selected="resource_selected"
+        )
+        self.resource.choices = []
+
+        self.table.headers_avail = ["Title", "Instrument", "DateObs", "URL"]
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
+        self._populate_url_only = False
+
+        self.table.show_rowselect = True
+        self.table.item_key = "URL"
+
+        self.hub.subscribe(self, AddDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, RemoveDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, LinkUpdatedMessage, handler=self.vue_center_on_data)
+
+    @property
+    def user_api(self):
+        return PluginUserApi(
+            self,
+            expose=(
+                # Traitlets
+                "source",
+                "viewer",
+                "radius",
+                "radius_unit",
+                "waveband",
+                "coordframe",
+                "resource_filter_coverage",
+                "resource",
+                # Methods
+                "center_on_data",
+                "query_registry_resources",
+                "load_selected_data",
+            ),
+        )
+
+    @observe("viewer_selected", type="change")
+    def vue_viewer_changed(self, _=None):
+        # Check mixin object initialized
+        if not hasattr(self, "viewer"):
+            return
+
+        # Clear all existing subscriptions and resubscribe to selected viewer
+        for viewer in self.viewer.viewers:
+            if viewer == self.viewer.selected_obj:
+                viewer.state.add_callback("zoom_center_x", self.vue_center_on_data)
+                viewer.state.add_callback("zoom_center_y", self.vue_center_on_data)
+            else:
+                # If not subscribed anyways, remove_callback should produce a no-op
+                viewer.state.remove_callback("zoom_center_x", self.vue_center_on_data)
+                viewer.state.remove_callback("zoom_center_y", self.vue_center_on_data)
+        self.vue_center_on_data()
+
+    @observe("coord_follow_viewer_pan", type="change")
+    def _toggle_viewer_pan_tracking(self, _=None):
+        """Detects when live viewer tracking toggle is clicked and centers on data if necessary"""
+        # Center on data if we're enabling the toggle
+        if self.coord_follow_viewer_pan:
+            self.vue_center_on_data()
+
+    def vue_center_on_data(self, event=None):
+        """UI entrypoint for the manual viewer center button"""
+        # If plugin is in "Manual" mode, we should never
+        # autocenter and potentially wipe the user's data
+        if not self.viewer_selected or self.viewer_selected == "Manual":
+            return
+
+        self.center_on_data(event)
+
+    def center_on_data(self, event=None):
```

Comment:
is it possible to do this without passing `event` (can you take out whatever you need from the event and pass as named kwargs that the user could pass manually)?

Reply:
Good point! Like this? https://github.com/spacetelescope/jdaviz/pull/2872/commits/859379fbdb0e106caaea7bb47bc7e706eb3b6320

Reply:
if you don't need anything from the event, can you just not pass it at all by calling `self.center_on_data()` and `def center_on_data(self):`?

Reply:
Oh right, duh. Good point! I dropped the vent arg here: https://github.com/spacetelescope/jdaviz/pull/2872/commits/844c72123db84d06565000cb585923babae64a47